go/stellar/stellar.go

```diff
@@ -2133,3 +2257,27 @@ func AirdropStatus(mctx libkb.MetaContext) (stellar1.AirdropStatus, error) {
 	}
 	return TransformToAirdropStatus(apiStatus), nil
 }
+
+func FindPaymentPath(mctx libkb.MetaContext, remoter remote.Remoter, source stellar1.AccountID, to string, sourceAsset, destinationAsset stellar1.Asset, amount string) (stellar1.PaymentPath, error) {
+	recipient, err := LookupRecipient(mctx, stellarcommon.RecipientInput(to), false)
+	if err != nil {
+		return stellar1.PaymentPath{}, err
+	}
+	if recipient.AccountID == nil {
+		return stellar1.PaymentPath{}, errors.New("cannot send a path payment to a user without a stellar account")
+	}
+
+	sourceEntry, _, err := LookupSender(mctx, source)
+	if err != nil {
+		return stellar1.PaymentPath{}, err
+	}
+
+	query := stellar1.PaymentPathQuery{
+		Source:           sourceEntry.AccountID,
+		Destination:      stellar1.AccountID(recipient.AccountID.String()),
+		SourceAsset:      sourceAsset,
+		DestinationAsset: destinationAsset,
+		Amount:           amount,
+	}
+	return remoter.FindPaymentPath(mctx, query)
```

Comment:
if the remote server swapped out same values like destination account could that trick the client?

Reply:
FindPaymentPath does not return the destination account.

If the remote server modified the argument to use a different destination account for the call to find the path, either the destination account would have a trustline to the asset or not.  That's the only difference.  Horizon doesn't even look at the destination account (I filed a ticket about this), but stellarnet checks that it has a trustline to the destination asset.  So worst case a malicious server would either return a path that wouldn't work when submitted or an error for a path that would work.  Neither of these would result in a loss of funds, just bad user experience on an advanced feature.

The call to SendPathPayment requires a recipient, which is looked up to get the account ID on the client at the time that the path payment transaction is constructed.